src/components/SignIn.tsx

```diff
@@ -189,116 +52,44 @@ const SignIn: FC = () => {
-                }
-              >
-                {t(SIGN_IN_BUTTON)}
-              </LoadingButton>
-            </>
-          )}
-          {signInMethod === SIGN_IN_METHODS.EMAIL && (
-            <>
-              <ErrorDisplay error={signInError} />
-              <LoadingButton
-                id={SIGN_IN_BUTTON_ID}
-                variant="contained"
-                onClick={handleSignIn}
-                loading={isLoadingMobileSignIn || isLoadingSignIn}
-              >
-                {t(SIGN_IN_BUTTON)}
-              </LoadingButton>
-            </>
-          )}
-        </Stack>
-      </FormControl>
-      <Link to={`${SIGN_UP_PATH}${search}`}>{t(SIGN_UP_LINK_TEXT)}</Link>
-    </>
-  );
-
   return (
     <FullscreenContainer>
       {
-        // eslint-disable-next-line no-constant-condition
-        (signInSuccess ||
-          signInWithPasswordSuccess ||
-          mobileSignInSuccess ||
-          mobileSignInWithPasswordSuccess) &&
-        successView ? (
-          <SuccessContent
-            title={t(AUTH.SIGN_IN_SUCCESS_TITLE)}
-            email={email}
-            handleBackButtonClick={handleBackButtonClick}
-          />
-        ) : (
-          <Stack direction="column" alignItems="center" spacing={2}>
-            <Stack spacing={1}>
-              <GraaspLogo
-                height={90}
-                sx={{ fill: theme.palette.primary.main }}
-              />
-              <Typography variant="h4" component="h2" id={SIGN_IN_HEADER_ID}>
-                {t(SIGN_IN_HEADER)}
-              </Typography>
-            </Stack>
-            {renderSignInForm()}
-            <Box justifyContent="center">
-              <Button
-                variant="text"
-                disabled={signInMethod === SIGN_IN_METHODS.EMAIL}
-                onClick={handleSignInMethod}
-                id={EMAIL_SIGN_IN_METHOD_BUTTON_ID}
-              >
-                {t(EMAIL_SIGN_IN_METHOD)}
-              </Button>
-              <Button
-                variant="text"
-                disabled={signInMethod === SIGN_IN_METHODS.PASSWORD}
-                onClick={handleSignInMethod}
-                id={PASSWORD_SIGN_IN_METHOD_BUTTON_ID}
-              >
-                {t(PASSWORD_SIGN_IN_METHOD)}
-              </Button>
-            </Box>
+        <Stack direction="column" alignItems="center" spacing={2}>
+          <Stack spacing={1}>
+            <GraaspLogo height={90} sx={{ fill: theme.palette.primary.main }} />
+            <Typography variant="h4" component="h2" id={SIGN_IN_HEADER_ID}>
+              {t(SIGN_IN_HEADER)}
+            </Typography>
           </Stack>
-        )
+          <FormControl>
+            <Stack direction="column" spacing={1}>
+              {signInMethod === SIGN_IN_METHODS.PASSWORD && (
+                <SignInPasswordForm />
+              )}
+              {signInMethod === SIGN_IN_METHODS.EMAIL && <MagicLinkForm />}
```

Comment:
I think something like that should work ? 
```suggestion
              {signInMethod === SIGN_IN_METHODS.PASSWORD ? 
                <SignInPasswordForm /> : <MagicLinkForm />}
```
```suggestion
              {signInMethod === SIGN_IN_METHODS.PASSWORD && (
                <SignInPasswordForm />
              )}
              {signInMethod === SIGN_IN_METHODS.EMAIL && <MagicLinkForm />}
```